ctrlf.el

```diff
@@ -884,18 +895,30 @@ change the search style if already in a search."
     (unless preserve
       (setq ctrlf--style style))
     (if ctrlf--active-p
-        (if (and (not inhibit-history)
-                 (string-empty-p (field-string (point-max))))
-            (previous-history-element 1)
-          (ctrlf-next-match))
+        (cond
+         ;; Continue search with `arg'.
+         ;;
+         ;; TODO: Is there a better way to replace the minibuffer
+         ;; contents? Should we use `(with-current-buffer
+         ;; ctrlf--minibuffer...)' or is this safe?
+         (arg
+          (delete-minibuffer-contents)
+          (insert arg))
+         ;; Insert the previous search string.
+         ((and (not inhibit-history)
+               (string-empty-p (field-string (point-max))))
+          (previous-history-element 1))
+         ;; Go to next candidate.
+         (t (ctrlf-next-match)))
       (setq ctrlf--backward-p nil)
-      (ctrlf--start))))
+      (ctrlf--start arg))))
```

Comment:
I'm wondering if I should just pass `pos` through here too, but it doesn't make much other sense for `ctrlf-forward`. There are two main scenarios where we'd call `ctrlf-forward` from `ctrlf-forward-symbol-at-point`.

1. Hit `C-s`, enter no input, realize you want symbol under point, hit `M-s .` to start searching for it.
2. Be in the middle of a search, partially within a symbol, desire to search for the rest of the symbol, hit `M-s .` to fill it in. (Kind of like `C-w`, which I haven't implemented yet, but for the full symbol.)